subsys/bluetooth/host/att.c

```diff
@@ -612,7 +612,19 @@ static u8_t find_type_cb(const struct bt_gatt_attr *attr, void *user_data)
 	}
 
 	/* Check if data matches */
-	if (read != data->value_len || memcmp(data->value, uuid, read)) {
+	if (read != data->value_len) {
+		/* Use bt_uuid_cmp() to compare UUIDs of different form. */
+		struct bt_uuid_128 ref_uuid;
+		struct bt_uuid_128 recvd_uuid;
+
+		bt_uuid_create_le(&recvd_uuid.uuid, data->value,
```

Comment:
Return value not checked. Do we need to handle this?

Reply:
Checks are added in next commit